frictionless/resource/resource.py

```diff
@@ -619,8 +619,112 @@ def validate(
             Report: validation report
 
         """
-        validator = Validator()
-        return validator.validate_resource(self, checklist=checklist)
+        # Create state
+        partial = False
+        timer = helpers.Timer()
+        labels: List[str] = []
+        errors: List[Error] = []
+        warnings: List[str] = []
+
+        # Prepare checklist
+        checklist = checklist or Checklist()
+        checks = checklist.connect(self)
+
+        # Validate metadata
+        try:
+            self.to_descriptor(validate=True)
+        except FrictionlessException as exception:
+            return Report.from_validation_task(
+                self, time=timer.time, errors=exception.to_errors()
+            )
+
+        # TODO: remove in next version
```

Comment:
I usually like to add more explicit messages for future maintainers. 

Something like:
- `Remove in X.Y.Z when foo gets deprecated`
- `Remove when Y is removed`
- `Remove when we no longer support X`

Reply:
Yes indeed ! (This was copy-pasted). 
I think this is a breaking change from v4 to v5, so let's say  "remove in v6".
(I'll check for any dependency to the hash, but to me this feature is dropped with no dependencies)